server/src/shared/infra/cache/redis-repository.ts

```diff
@@ -0,0 +1,64 @@
+import { RedisClient } from "../../../setup/cache/redis"
+import { AppError } from "../../core/app-error"
+import { Result } from "../../core/result"
+import { RedisEntity } from "./redis-entity"
+
+export type RedisSaveEntityResponse = Result<RedisSaveEntitySuccess, RedisSaveEntityError>
+export type RedisSaveEntitySuccess = boolean
```

Comment:
From the way you use `RedisSaveEntityResponse`, it looks like `RedisSaveEntitySuccess` should never actually be `false`.

```suggestion
export type RedisSaveEntitySuccess = true
```